mne/preprocessing/tests/test_realign.py

```diff
@@ -45,45 +52,87 @@ def test_realign(ratio_other, start_raw, start_other, stop_raw, stop_other):
     data_raw = np.array(
         [
             interp1d(times, d, kind)(raw_times)
-            for d, kind in ((signal, "linear"), (stim, "nearest"))
+            for d, kind in (
+                (stim, "nearest"),
+                (sig_hann, "linear"),
+                (sig_box, "linear"),
+            )
         ]
     )
     data_other = np.array(
         [
             interp1d(times, d, kind)(other_times)
-            for d, kind in ((signal, "linear"), (stim, "nearest"))
+            for d, kind in (
+                (stim, "nearest"),
+                (sig_hann, "linear"),
+                (sig_box, "linear"),
+            )
         ]
     )
-    info_raw = create_info(["raw_data", "raw_stim"], sfreq, ["eeg", "stim"])
-    info_other = create_info(["other_data", "other_stim"], sfreq, ["eeg", "stim"])
-    raw = RawArray(data_raw, info_raw, first_samp=111)
+    info_raw = create_info(
+        ["raw_stim", "raw_hann", "raw_box"], sfreq, ["stim", "eeg", "eeg"]
+    )
+    info_other = create_info(
+        ["other_stim", "other_hann", "other_box"], sfreq, ["stim", "eeg", "eeg"]
+    )
+    raw = RawArray(data_raw, info_raw, first_samp=111)  # first_samp doesn't matter
     other = RawArray(data_other, info_other, first_samp=222)
 
-    # naive processing
     evoked_raw, events_raw, _, events_other = _assert_similarity(raw, other, n_events)
-    if start_raw == start_other:  # can just naively crop
-        a, b = data_raw[0], data_other[0]
-        n = min(len(a), len(b))
-        corr = np.corrcoef(a[:n], b[:n])[0, 1]
-        min_, max_ = (0.99999, 1.0) if sfreq_raw == sfreq_other else (0.8, 0.9)
-        assert min_ <= corr <= max_
```

Comment:
I think we should try to keep this in if we can... you say this is discarded because it only works when there is not much drift.  Do the failures with larger drift still happen if you use "nearest" interpolation on `sig_box`?

Reply:
Yes the failure persists with nearest interpolation when drift is large. TBH I don't think the test makes much sense. It asserts that `raw` and `other` should contain somehow correlated data, which is clearly violated when there's drift. While `min_` and `_max` were adjusted accordingly in that case, the range was somewhat arbitrary (0.8 and 0.9)